### PR TITLE
DM-40303: Minor clean ups for pydantic v2

### PR DESCRIPTION
**Pull request by timj**
## Checklist

- [x] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/ctrl/mpexec/cmdLineFwk.py [hunk 1]
```diff
@@ -250,7 +251,7 @@ def check(self, args: SimpleNamespace) -> None:
             # Passing the same inputs that were used to initialize the output
             # collection is allowed; this means they must _end_ with the same
             # collections, because we push new runs to the front of the chain.
-            for c1, c2 in zip(self.inputs[::-1], self.output.chain[::-1]):
+            for c1, c2 in zip(self.inputs[::-1], self.output.chain[::-1], strict=True):
```
Comment (timj) [inline python/lsst/ctrl/mpexec/cmdLineFwk.py:254]: This one also breaks everything...

File: python/lsst/ctrl/mpexec/execFixupDataId.py [hunk 2]
```diff
@@ -108,16 +109,15 @@ def fixupQuanta(self, graph: QuantumGraph) -> QuantumGraph:
         keys = sorted(keyQuanta.keys(), reverse=self.reverse)
         networkGraph = graph.graph
 
-        for prev_key, key in zip(keys, keys[1:]):
+        for prev_key, key in zip(keys, keys[1:], strict=True):
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply python/lsst/ctrl/mpexec/execFixupDataId.py:112]: Right, this one iterates over pairs of consecutive items in the same list. You should be able to fix it with

    for prev_key, key in zip(keys[:-1], keys[1:], strict=True):

File: python/lsst/ctrl/mpexec/cmdLineFwk.py (hunk 1 above)
Comment (andy-slac) [reply python/lsst/ctrl/mpexec/cmdLineFwk.py:254]: I think `strict=False` is the right thing to do here.

File: python/lsst/ctrl/mpexec/execFixupDataId.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (andy-slac) [reply python/lsst/ctrl/mpexec/execFixupDataId.py:112]: And it should probably use `itertools.pairwise` instead.

**Review by andy-slac — APPROVED** [review]
Looks good.

**Pull request merged**